### PR TITLE
SONARJAVA-1025 Nested enums are implicitly static

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

File: java-squid/src/test/java/org/sonar/java/resolve/SymbolTableTest.java [hunk 1]
```diff
@@ -253,7 +253,7 @@ public void InterfaceDeclaration() {
 
     typeSymbol = (JavaSymbol.TypeJavaSymbol) result.symbol("NestedEnum");
     assertThat(typeSymbol.owner()).isSameAs(interfaceSymbol);
-    assertThat(typeSymbol.flags()).isEqualTo(Flags.PUBLIC | Flags.ENUM);
+    assertThat(typeSymbol.flags()).isEqualTo(Flags.PUBLIC | Flags.ENUM | Flags.STATIC);
```
Comment (benzonico) [inline java-squid/src/test/java/org/sonar/java/resolve/SymbolTableTest.java]: Is there some tests to verify that an enum is NOT static in some cases ?

**Comment by benzonico** [comment]
See comment, otherwise LGTM

File: java-squid/src/test/java/org/sonar/java/resolve/SymbolTableTest.java (hunk 1 above)
Comment (m-g-sonar) [inline java-squid/src/test/java/org/sonar/java/resolve/SymbolTableTest.java]: No, there is no such test. I'll add them.

**Pull request closed without merging**